clients/src/main/java/org/apache/kafka/clients/InFlightRequests.java

```diff
@@ -158,7 +158,9 @@ public Iterable<NetworkClient.InFlightRequest> clearAll(String node) {
 
     private Boolean hasExpiredRequest(long now, Deque<NetworkClient.InFlightRequest> deque) {
         for (NetworkClient.InFlightRequest request : deque) {
-            if (request.timeElapsedSinceSendMs(now) > request.requestTimeoutMs)
+            // We exclude throttle time here because we want to ensure that we don't expire requests while
+            // they are throttled. The request timeout should take effect only after the throttle time has elapsed.
+            if (request.timeElapsedSinceSendMs(now) - request.throttleTimeMs() > request.requestTimeoutMs)
                 return true;
```

Comment:
one minor tricky thing here is that we never have a guarantee that we actually throttled fro `throttleTimeMs` precisely, because:
- `time.milliseconds()` isn't monotonic
- we use `time.milliseconds()` to compute the end time for throttling

But this is equally inconsistent with the other two things here - `request.timeElapsedSinceSendMs(now)` and `request.requestTimeoutMs` - so I guess that's fine. Not sure why Kafka doesn't use hiResClockMs here

Reply:
probably because we attach the timeoutMs to the request?